### PR TITLE
Add Brakeman

### LINKED ISSUE #728: add brakeman security scanner
https://brakemanscanner.org/

### PR DESCRIPTION
**Pull request by haydenrou**
This PR adds [Brakeman](https://github.com/presidentbeef/brakeman). Brakeman identifies potential security vulnerabilities.

A github workflow has also been added to identify any issues on pull requests.

Closes #728

### CODE REVIEW

**Comment by haydenrou** [comment]
Workflows will pass after https://github.com/rubyforgood/Flaredown/pull/732 has been merged

File: backend/Gemfile [hunk 1]
```diff
@@ -12,6 +12,9 @@ gem "rake"
 # JSON serializer
 gem "active_model_serializers", "0.9.8"
 
+# Brakeman security scanner
+gem "brakeman", "6.1.2"
```
Comment (Tonkpils) [inline backend/Gemfile]: Should this be a development dependency only? I don't think we want this installed in production.

Comment (haydenrou) [reply backend/Gemfile]: Good point!

**Review by Tonkpils — APPROVED** [review]

**Comment by compwron** [comment]
great! :)

**Pull request merged**